### PR TITLE
bpf: nat: minor improvements

### PR DESCRIPTION
**Pull request by julianwiedmann**
Some small housekeeping patches for our NAT code
- remove dead code  / comments
- avoid accidental inclusion of some NAT code into bpf_overlay

### CODE REVIEW

**Comment by qmonnet** [comment]
This will need a rebase on top of https://github.com/cilium/cilium/pull/26501.

Looks good!

File: bpf/lib/nat.h [hunk 1]
```diff
@@ -742,7 +743,9 @@ snat_v4_prepare_state(struct __ctx_buff *ctx, struct ipv4_nat_target *target)
 		return NAT_NEEDED;
 	}
 # endif
-#else
+#endif /* TUNNEL_MODE && IS_BPF_OVERLAY */
+
+#if defined(IS_BPF_HOST)
```
Comment (qmonnet) [inline bpf/lib/nat.h:748]: Nit:

```suggestion
#ifdef IS_BPF_HOST
```

(and same below)

**Review by qmonnet — APPROVED** [review]
Seems all good to me, as far as I can tell. Just one nit below.

Thanks for removing the comment I forgot to clean-up :facepalm:

**Comment by qmonnet** [comment]
Removing the label to leave you a chance to address the nit, if you want to.

**Comment by julianwiedmann** [comment]
> Removing the label to leave you a chance to address the nit, if you want to.

Thanks for the super-fast review! Think I'll save myself the extra round through CI :)

**Pull request merged**